opensrp-reveal/src/main/java/org/smartregister/reveal/adapter/ChildRegisterAdapter.java

```diff
@@ -0,0 +1,120 @@
+package org.smartregister.reveal.adapter;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import org.apache.commons.lang3.StringUtils;
+import org.smartregister.reveal.R;
+import org.smartregister.reveal.model.Child;
+import org.smartregister.reveal.util.Constants;
+import org.smartregister.reveal.viewholder.GroupedListableViewHolder;
+import org.smartregister.view.ListContract;
+
+import java.util.List;
+
+public class ChildRegisterAdapter extends GroupedListableAdapter<Child, GroupedListableViewHolder<Child>> {
+
+    public ChildRegisterAdapter(List<Child> items, ListContract.View<Child> view) {
+        super(items, view);
+    }
+
+    @Override
+    public void reloadData(@Nullable List<Child> items) {
+        super.reloadData(items);
+    }
+
+    @NonNull
+    @Override
+    public GroupedListableViewHolder<Child> onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.child_register_list_row, parent, false);
+        return new ChildViewHolder(view);
+    }
+
+    public static class ChildViewHolder extends GroupedListableViewHolder<Child> {
+
+        private TextView tvName;
+        private TextView tvDetails;
+        private View currentView;
+        private View linearLayoutAction;
+        private TextView tvHeader;
+        private Button btnDue;
+        private Context context;
+
+        private ChildViewHolder(@NonNull View itemView) {
+            super(itemView);
+            currentView = itemView;
+            tvHeader = itemView.findViewById(R.id.tvHeader);
+            tvName = itemView.findViewById(R.id.patient_name);
+            tvDetails = itemView.findViewById(R.id.tvDetails);
+            linearLayoutAction = itemView.findViewById(R.id.linearLayoutAction);
+            btnDue = itemView.findViewById(R.id.btnDue);
+            context = itemView.getContext();
+        }
+
+        @Override
+        public void bindView(Child child, ListContract.View<Child> view) {
+            tvName.setText(child.getFullName());
+            String details = "#" + child.getUniqueID() + " \u00B7 " + child.getGender() + " \u00B7 Age " + child.getAge();
+            tvDetails.setText(details);
+            currentView.setOnClickListener(v -> view.onListItemClicked(child, v.getId()));
+            btnDue.setOnClickListener(v -> view.onListItemClicked(child, v.getId()));
+
+            if (StringUtils.isBlank(child.getTaskStatus()))
+                return;
+
+            switch (child.getTaskStatus()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5414138)